lib/util/native/autostart_helper.dart

```diff
@@ -2,39 +2,67 @@ import 'dart:io';
 
 import 'package:launch_at_startup/launch_at_startup.dart';
 import 'package:localsend_app/init.dart';
+import 'package:localsend_app/model/state/settings_state.dart';
 import 'package:logging/logging.dart';
 import 'package:package_info_plus/package_info_plus.dart';
 import 'package:url_launcher/url_launcher.dart';
 
 final _logger = Logger('AutoStartHelper');
 
 /// Currently, only works for windows
-Future<void> initAutoStartAndOpenSettings() async {
+Future<bool> initEnableAutoStartAndOpenSettings(SettingsState settings, [bool? isWindows]) async {
   try {
     // In case somebody don't use msix
     final packageInfo = await PackageInfo.fromPlatform();
     launchAtStartup.setup(
       appName: packageInfo.appName,
       appPath: Platform.resolvedExecutable,
-      args: [launchAtStartupArg],
+      args: [if (!settings.autoStartLaunchMinimized) launchAtStartupArg],
```

Comment:
So you call `initDisableAutoStart` and `initEnableAutoStartAndOpenSettings` for the "launch minimized" setting only to update the `launchAtStartupArg` flag right?

Reply:
Yes, I ensured the deletion of the .desktop file / registry before writing it again

Reply:
I don't know if this is needed as we already check the setting during startup: https://github.com/localsend/localsend/blob/d8babe05970066536162054511265e40013ad72b/lib/init.dart#L100

Reply:
This check is only relevant for the initial boot of the App, but when writing the argument to the File / Registry it needs to follow the current settings.autoStartLaunchMinimized value, otherwise, it will always be written.

Reply:
> but when writing the argument to the File / Registry it needs to follow the current settings.autoStartLaunchMinimized value

Can you elaborate the other scenarios? Isn't `autoStartLaunchMinimized` only used for the initial boot of the app?
Assuming, `launchAtStartup` is `true`, `autoStartLaunchMinimized` is`false` and the `launchAtStartupArg` flag is enabled. The app should start normaly (not minimized) as expected as `autoStartLaunchMinimized` is `false` during init.

From my perspective, the flag `launchAtStartupArg` only tells LocalSend that the app was started during boot / login. The "launch minimized" feature is independent from this flag and only a feature built on top.

Reply:
Well, the changes made the `!persistenceService.isAutoStartLaunchMinimized()` check redundant indeed. Now that the autostart flag accurately reflects the user's choice. Tbh I forgot that this check existed until you pointed it out 😆, but I suppose that it does work both ways. The only difference is that the autostart argument will be written only when requested by the user.

Reply:
Well, as long as it works :smile: